pandas/_libs/tslib.pyx

```diff
@@ -515,9 +515,10 @@ cpdef array_to_datetime(
                             iresult[i] = NPY_NAT
                             continue
                         elif is_raise:
+                            match_msg = "match" if exact else "search"
                             raise ValueError(
                                 f"time data \"{val}\" at position {i} doesn't "
-                                f"match format \"{format}\""
+                                f"match format \"{format}\" ({match_msg})"
```

Comment:
ending this with "search" seems weird (and i dont see it in any test cases)

Reply:
It's true that it wasn't tested, but it's been like this since pandas 0.24:
```python
In [1]: import pandas

In [2]: pandas.__version__
Out[2]: '0.24.0'

In [3]: pandas.to_datetime(['2020-01-f00'], format='%Y-%d-%m', exact=False)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/tmp/.venv37/lib/python3.7/site-packages/pandas/core/tools/datetimes.py in _convert_listlike_datetimes(arg, box, format, name, tz, unit, errors, infer_datetime_format, dayfirst, yearfirst, exact)
    290             try:
--> 291                 values, tz = conversion.datetime_to_datetime64(arg)
    292                 return DatetimeIndex._simple_new(values, name=name, tz=tz)

pandas/_libs/tslibs/conversion.pyx in pandas._libs.tslibs.conversion.datetime_to_datetime64()

TypeError: Unrecognized value type: <class 'str'>

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-3-8c364de9721c> in <module>
----> 1 pandas.to_datetime(['2020-01-f00'], format='%Y-%d-%m', exact=False)

~/tmp/.venv37/lib/python3.7/site-packages/pandas/core/tools/datetimes.py in to_datetime(arg, errors, dayfirst, yearfirst, utc, box, format, exact, unit, infer_datetime_format, origin, cache)
    607             result = _convert_and_box_cache(arg, cache_array, box, errors)
    608         else:
--> 609             result = convert_listlike(arg, box, format)
    610     else:
    611         result = convert_listlike(np.array([arg]), box, format)[0]

~/tmp/.venv37/lib/python3.7/site-packages/pandas/core/tools/datetimes.py in _convert_listlike_datetimes(arg, box, format, name, tz, unit, errors, infer_datetime_format, dayfirst, yearfirst, exact)
    292                 return DatetimeIndex._simple_new(values, name=name, tz=tz)
    293             except (ValueError, TypeError):
--> 294                 raise e
    295
    296     if result is None:

~/tmp/.venv37/lib/python3.7/site-packages/pandas/core/tools/datetimes.py in _convert_listlike_datetimes(arg, box, format, name, tz, unit, errors, infer_datetime_format, dayfirst, yearfirst, exact)
    259                 try:
    260                     result, timezones = array_strptime(
--> 261                         arg, format, exact=exact, errors=errors)
    262                     if '%Z' in format or '%z' in format:
    263                         return _return_parsed_timezone_results(

pandas/_libs/tslibs/strptime.pyx in pandas._libs.tslibs.strptime.array_strptime()

ValueError: time data '2020-01-f00' does not match format '%Y-%d-%m' (search)
```

Maybe we can just remove it - people typically know whether they've passed `exact`, and it doesn't really add much